### ISSUE TITLE
Crash on Android 7.1.2 that Log tag exceeds limit of 23 characters

### ISSUE THREAD

**Issue by mvysny** [mvysny/slf4j-handroid#2]
Typical: the Android API specification is one thing, the implementation of some vendor (ASUS in this case) is something completely different. Quoting `Log.isLoggable()` javadoc:

> IllegalArgumentException is thrown if the tag.length() > 23 for Nougat (7.0) releases (API <= 23) and prior, there is no tag limit of concern after this API level.

Guess what? ASUS_Z00UD with Android 7.1.2 (that is, on fucking API level 25) just failed with:

```
Caused by java.lang.IllegalArgumentException: Log tag "AedictKDP:sk.baka.aedictkanjidrawpractice.util.android.SodIndex" exceeds limit of 23 characters

       at android.util.Log.isLoggable(Log.java)
       at org.slf4j.impl.AndroidLoggerAdapter.isLoggable(AndroidLoggerAdapter.java:543)
       at org.slf4j.impl.HandroidLoggerAdapter.isLoggable(HandroidLoggerAdapter.java:96)
       at org.slf4j.impl.AndroidLoggerAdapter.log(AndroidLoggerAdapter.java:537)
       at org.slf4j.impl.AndroidLoggerAdapter.info(AndroidLoggerAdapter.java:281)
       at org.slf4j.impl.HandroidLoggerAdapter.info(HandroidLoggerAdapter.java:23)
       at sk.baka.aedictkanjidrawpractice.util.android.SodIndex.<init>(SodIndex.kt:54)
       at sk.baka.aedictkanjidrawpractice.util.android.SodCache.<init>(SodCache.kt:27)
       at sk.baka.aedictkanjidrawpractice.util.android.SodCacheKt.<clinit>(SodCache.kt:82)
       at sk.baka.aedictkanjidrawpractice.util.android.SodCacheKt.getSodCache(SodCache.kt:82)
       at sk.baka.aedictkanjidrawpractice.MainActivity.showKanji(MainActivity.kt:47)
       at sk.baka.aedictkanjidrawpractice.MainActivity.onCreate(MainActivity.kt:70)
       at android.app.Activity.performCreate(Activity.java:6684)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1119)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2637)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2751)
       at android.app.ActivityThread.-wrap12(ActivityThread.java)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6186)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
```

Maybe in the fucked-up Android world, 25 is less or equal than 23. This is the typical example of Android fragmentation and how defensively one must do code in such environment. Don't anything for granted. What a load of shit.

**Comment by mvysny** [comment]
And there is nothing to do besides crafting a workaround. Asus will probably never fix issues like this one. I'm really happy that I don't have to do Android coding as my job.

**Comment by consp1racy** [comment]
Yeah, looks like Android has a, umm, "fucked-up" documentation.

> for Nougat (7.0) releases (API <= 23) and prior, there is no tag limit of concern after this API level.

The note was added for Oreo (Android 8) release and the limit was dropped then.

Here's what I was able to dig up: https://stackoverflow.com/a/54744028/2444099

I reported it here https://issuetracker.google.com/issues/124593220

Hopefully we get the documentation fixed for next SDK release... Now I know of two logging libraries that got screwed by this poor documentation.
